doc(.github): add pr template (#325)
Qytera-Gmbh/QTAF: .github/.github/pull_request_template.md

Reviewer: Should the file's path really be `.github/.github/pull_request_template.md`?

Author: No, you're right, it should just be `.github/pull_request_template.md`. Take a look at the open discussions and when those are closed, I'll adjust that. (I'm not sure what will happen to the pending suggestions if I move the file now)

Author: I moved the file into the correct directory now.

# 📄 PR Description

> [Add a description for this pull request. Follow the checklist below.] 

Reviewer: ```suggestion
<!--
Add a description for this pull request. Follow the checklist below.
- What is the purpose of the pull request?
- Illustrate the problem this PR solves and why this solution is appropriate.
  - Include an overview of the changes, ideally with 1-2 examples that illustrate the changes.
  - Include links to any additional context such as tracking issues or tickets.
-->
```

<!--- Describe your changes in detail including a relevant motivation and context for this change -->

> # ✒️ Checkliste PR-Author

Reviewer: ```suggestion
# ✒️ Checklist PR author
```

## Before submitting the PR, please make sure you do the following

### Prepare Code for PR

> * [ ] The automated tests were executed locally (`mvn clean test`), and it was ensured that they passed.

Reviewer: ```suggestion
* [ ] The automated tests were successfully executed locally (`mvn clean test`).
```

* [ ] All commented-out code and debug statements have been removed.
* [ ] A test is included that fails without this PR but passes with it.

> * [ ] The relevant documentation, including code comments and public documentation, has been updated.

Author: ```suggestion
* [ ] The documentation in the code has been updated and corresponds to the team guidelines
* [ ] A person has been assigned to update the public documentation as soon as possible if this PR requires changes to it
```

What do you think about this instead of the suggestion above?

> * [ ] The relevant documentation, including code comments and public documentation, has been updated.

Reviewer: ```suggestion
* [ ] The relevant documentation, including code comments and public documentation, has been updated in a corresponding PR.
```

I think it does not make sense to update the productive documentation before the changes actually go live.

Author: Yes, I can partly understand that. But in my opinion, it definitely makes sense to update the documentation in the code. I think the public documentation should also be adapted as soon as possible. However, is it difficult to guarantee that the check will be carried out later. What do you think?
You can't guarantee at this point that the documentation will be adapted at a late stage. The idea here was to only allow changes to the production code once the documentation has also been adapted. But it true, now you might create a public documentation that is ahead of the actual code, which wouldn't be good either.

Author: ```suggestion
* [ ] The documentation in the code has been updated and corresponds to the team guidelines.
* [ ] A person has been assigned to update the public documentation as soon as possible if this PR requires changes to it.
```

What do you think about this instead of the suggestion above?

Reviewer: > But it true, now you might create a public documentation that is ahead of the actual code, which wouldn't be good either.

I usually create a new PR in the documentation repository and merge them when the new version is released as well.

> What do you think about this instead of the suggestion above?

That's better!

    * [ ] [Qytera Github-Page (github.io)](https://github.com/Qytera-Gmbh/Qytera-Gmbh.github.io)

> ### PR Description
> * [ ] A message body is added that provides context and guidance so that the reviewer can quickly understand what the PR does. In the PR body, this includes. In the PR body, includes:
>     * [ ] The purpose of the pull request, clearly illustrating what problem this PR solves and why this solution is appropriate.
>     * [ ] An overview of the changes, ideally with 1-2 examples that illustrate the changes.
>     * [ ] Links to any additional context such as tracking issues or tickets.

Reviewer: ```suggestion
```

I'd simply include that in the description above (see https://github.com/Qytera-Gmbh/QTAF/pull/325#discussion_r1679122687).

### Review Preparation
* [ ] Review from appropriate team members is requested.

> - Be prepared to explain and discuss changes.

Reviewer: ```suggestion
```

I think this is unnecessary. It doesn't add any value (you wouldn't create a PR if you weren't ready for feedback).

> # 🔍 Checkliste PR-Reviewer

Reviewer: ```suggestion
# 🔍 Checklist  PR reviewer
```

## Before accepting the PR, please make sure you do the following

### Check Preconditions
* [ ] It is confirmed that the author has fulfilled all preconditions. (The review can only take place once this has been fulfilled.)

### Understand the Context
* [ ] The PR description is read thoroughly to understand the changes.
* [ ] It is verified that the changes fulfill the purpose described in the PR.

> * [ ] If the PR belongs to an issue or is intended to represent a user story, it is verified that the requirements and acceptance criteria are met. (If necessary, the author is asked where these were specified.)

Reviewer: ```suggestion
* [ ] If the PR belongs to an issue or is intended to represent a user story, it is verified that the requirements and acceptance criteria are met.
```

Redundant with the subsequent bullet point.

- If things are unclear, please ask the author to add or adjust them in the description or as a comment. 

### Code Quality, Style & Functionality
* [ ] It is ensured that the code adheres to the projects' [coding standards and style guides](https://qytera.atlassian.net/wiki/spaces/QO/pages/3800727592/QTAF+-+Coding+Standards).

> * [ ] The code is clean, readable and well-organized code. 

Reviewer: ```suggestion
* [ ] The code is clean, readable and well-organized. 
```

> * [ ] The commits are named according to [the team conventions](https://qytera.atlassian.net/wiki/spaces/WIS/pages/3861348360/Commit+und+PR+Konventionen+f+r+QTAF). (If not, the author is informed so that he/she can improve for the next time. There are no changes to the commit history required.)

Reviewer: This is an internal link whose contentshould probably be moved to a [`CONTRIBUTING.md`](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors) of some sort. But since nobody but Qytera is developing QTAF anyways, I think this is fine for now.

Author: Yes, that is correct. I also think that it will work for now.

* [ ] It is ensured that there are no commented-out code or debug statements left.

> * [ ] Potential edge cases and error handling are checked. (Are potential errors properly retained? Is there an error handling strategy?)

Reviewer: ```suggestion
* [ ] The changed or added code correctly handles potential edge cases and errors.
```
